### ISSUE TITLE
Bukkit attempting to fix blocks, console spam

### ISSUE THREAD

**Issue by Vampsku11** [PaperMC/Paper#2680]
minecraft_server.1.14.4.jar
paper-217.jar and up

With Tyruswoo Game Changer datapack installed (requires patreon donation), since updating paper to 217 or newer I am getting console spam. This does not happen in paper 216 or older.

The datapack generates structures and uses command blocks to do so. When new chunks are generated and a structure is generated within them, the errors occur.

Getting many logs similar to these:
```
[19:06:20 ERROR]: Block at 227, 38, -931 is Block{minecraft:stone} but has net.minecraft.server.v1_14_R1.TileEntityStructure@527ce791. Bukkit will attempt to fix this, but there may be additional damage that we cannot recover.
[19:06:20 ERROR]: Block at 266, 38, -970 is Block{minecraft:barrier} but has net.minecraft.server.v1_14_R1.TileEntityStructure@71c5dc05. Bukkit will attempt to fix this, but there may be additional damage that we cannot recover.
[19:06:20 ERROR]: Block at 240, 10, -944 is Block{minecraft:stone} but has net.minecraft.server.v1_14_R1.TileEntityStructure@5a3c2288. Bukkit will attempt to fix this, but there may be additional damage that we cannot recover.
[19:06:20 ERROR]: Block at 214, 38, -931 is Block{minecraft:polished_diorite} but has net.minecraft.server.v1_14_R1.TileEntityStructure@1c928bc6. Bukkit will attempt to fix this, but there may be additional damage that we cannot recover.
[19:06:20 ERROR]: Block at 253, 38, -931 is Block{minecraft:polished_diorite} but has net.minecraft.server.v1_14_R1.TileEntityStructure@2701c2be. Bukkit will attempt to fix this, but there may be additional damage that we cannot recover.
[19:06:20 ERROR]: Block at 227, 31, -944 is Block{minecraft:stone} but has net.minecraft.server.v1_14_R1.TileEntityStructure@70ef6c25. Bukkit will attempt to fix this, but there may be additional damage that we cannot recover.
[19:06:20 ERROR]: Block at 214, 31, -944 is Block{minecraft:stone} but has net.minecraft.server.v1_14_R1.TileEntityStructure@422424d7. Bukkit will attempt to fix this, but there may be additional damage that we cannot recover.
[19:06:20 ERROR]: Block at 227, 38, -944 is Block{minecraft:dark_oak_fence} but has net.minecraft.server.v1_14_R1.TileEntityStructure@44f9f2f7. Bukkit will attempt to fix this, but there may be additional damage that we cannot recover.
[19:06:20 ERROR]: Block at 227, 38, -918 is Block{minecraft:oak_planks} but has net.minecraft.server.v1_14_R1.TileEntityStructure@269f3c7. Bukkit will attempt to fix this, but there may be additional damage that we cannot recover.
[19:06:20 ERROR]: Block at 240, 10, -918 is Block{minecraft:chiseled_stone_bricks} but has net.minecraft.server.v1_14_R1.TileEntityStructure@90cd737. Bukkit will attempt to fix this, but there may be additional damage that we cannot recover.
```

I believe the errors are due to this change in 217:
CraftBukkit Changes:
e2dd6555 SPIGOT-5372: Re-add tile entity fixer due to MC-163945

I could find no obvious issues with the blocks themselves.

I could not find an option to disable the feature or suppress the logs. I am looking to prevent the console spam, if there is an option to do so now or if an option can be added.

**Comment by aikar** [comment]
https://www.spigotmc.org/resources/console-spam-fix.18410/
You can use this to hide logs.

**Comment by justEli** [comment]
I can confirm this happens since recently.

**Comment by Co2Noss** [comment]
It was happening a lot with WirelessRedstone plugin jar I noticed. I've had to temporarily remove the plugin and I've notified the developer as well but this issue just started after I updated to paper-225.jar. We were using 214 before.

**Comment by Vampsku11** [comment]
I did download every version from 216 to 225 and tried every one until I found the issue began specifically with 217 and up. Can you guys confirm if version 217 also gives the errors but 216 does not?

**Comment by Spottedleaf** [comment]
The system introduced to fix issues was introduced in that build (217). It's not like 217 caused anything, but rather the build simply starting logging when this was occurring.

**Comment by Vampsku11** [comment]
The issue is that the console is spammed. The logs either need to be suppressed or the feature disabled to prevent logs.

Also could this potentially cause problems with the structures that are being generated? Were these errors attempted to be fixed before or only after 217 and can that affect gameplay?

**Comment by electronicboy** [comment]
An option has already been given to suppress the logs, ultimately, the underlying issue needs to be resolved to prevent your world being in a broken state where these errors are thrown, but; as a note, this is not a new issue, just now there is logic to actually pick these broken states up;

Two major issues here is that;
1) How is the world ended up in these states in the first place
2) from a quick glance I had a while back, looked like the fix logic which won’t cover these specific cases, i.e. a TE exists where it shouldn't

**Comment by aeonlamb** [comment]
I've been getting this issue too. For me, it happens when my datapack clones some dispensers that have stuff inside them. Just the vanilla clone command does it. I can't see anything immediately wrong with the dispensers or anything, they have custom names and I enabled the locked nbt tag on them but that's about it. The chunk doesn't appear to be corrupted or strange in any way, so the errors just seem annoying.

**Comment by aikar** [comment]
The Chunk isn't corrupted BECAUSE of the log lines you see, its telling you it fixed it for you.

There is no turning it off. the system is fixing an illegal/corrupt state for you, and it's letting you know it happened.

If you don't want to know about it, and rather it happen silently, use the log suppressing plugin I linked.

**Comment by aeonlamb** [comment]
But why does the vanilla clone command cause the tile entity to become corrupted? Is that a mojang bug? Also I'd rather not suppress the warnings in case they happen in other places where I don't expect it to occur.

**Comment by electronicboy** [comment]
> Is that a mojang bug

Some causes of that error, yes; Other causes, "needs investigation"

**Comment by ghost** [comment]
> The Chunk isn't corrupted BECAUSE of the log lines you see, its telling you it fixed it for you.
> 
> There is no turning it off. the system is fixing an illegal/corrupt state for you, and it's letting you know it happened.
> 
> If you don't want to know about it, and rather it happen silently, use the log suppressing plugin I linked.
I have the same issue and I used the linked plugin it isnt working. pls help

**Comment by sammyt291** [comment]
I get this without using any plugins or datapacks, 
In creative if i place a trapped chest in water, i get this error each time i break a trapped chest.
